lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,427 @@
+    expect(inputElement).toHaveProperty('disabled', false);
+
+    rerender(<NxCombobox {...minimalProps} disabled={false} />);
+    expect(inputElement).toHaveProperty('disabled', false);
+
+    rerender(<NxCombobox {...minimalProps} matches={[{ id: '1', displayName: 'Foo' }]} disabled={false} />);
+    expect(getByRole('option', { hidden: true })).toHaveProperty('disabled', false);
+
+    rerender(<NxCombobox {...minimalProps} matches={[{ id: '1', displayName: 'Foo' }]} disabled={true} />);
+    expect(getByRole('option', { hidden: true })).toHaveProperty('disabled', true);
+  });
+
+  it('renders a dropdown with the nx-combobox__menu class', async function() {
+    const { getByRole } = rtlRender({ matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }] }),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveClass('nx-combobox__menu');
+  });
+
+  it('sets the listbox role on the dropdown', function() {
+    const { getByRole } = rtlRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveAttribute('role', 'listbox');
+  });
+
+  it('sets an id on the dropdown and references it in the combobox input aria-controls', function() {
+    const { getByRole } = rtlRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+        inputElement = getByRole('combobox'),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveAttribute('id');
+    expect(inputElement).toHaveAttribute('aria-controls', dropdownElement.id);
+  });
+
+  it('renders an alert dropdown with nx_combobox__alert class', function() {
+    const { container } = rtlRender(),
+        alertDropdownElement = container.querySelector('.nx-combobox__alert');
+
+    expect(alertDropdownElement).toBeTruthy();
+    expect(alertDropdownElement).toHaveClass('nx-combobox__alert');
+  });
+
+  it('sets the alert role on the alert dropdown when it is in loading, error, or empty states', function() {
+    expect(rtlRender({ matches: [] }).container.querySelector('.nx-combobox__alert')).toHaveAttribute('role', 'alert');
+    expect(rtlRender({ loading: true }).container.querySelector('.nx-combobox__alert'))
+        .toHaveAttribute('role', 'alert');
+    expect(rtlRender({ loadError: 'boo' }).container.querySelector('.nx-combobox__alert'))
+        .toHaveAttribute('role', 'alert');
+  });
+
+  it('sets an id on the alert dropdown and references it in the combobox input aria-describedby', function() {
+    const { getByRole, container } = rtlRender(),
+        inputElement = getByRole('combobox'),
+        alertDropdownElement = container.querySelector('.nx-combobox__alert')!;
+
+    expect(alertDropdownElement).toHaveAttribute('id');
+    expect(inputElement).toHaveAttribute('aria-describedby', alertDropdownElement.id);
```

Comment:
Use `toHaveAccessibleDescription` instead

Reply:
Updated here 4029d58f1778f46cf53cdedad783b595e4a9da4e.

Reply:
> Updated here https://github.com/sonatype/sonatype-react-shared-components/commit/4029d58f1778f46cf53cdedad783b595e4a9da4e.

Assert what the description is, and when it should and should not be present.  And that a caller-provided `aria-describedby` gets included as well

Reply:
> Assert what the description is, and when it should and should not be present. And that a caller-provided aria-describedby gets included as well

Updated 56541e1d60fd9a5530706547c3e5212d0f4ba3a3

Reply:
> Updated https://github.com/sonatype/sonatype-react-shared-components/commit/56541e1d60fd9a5530706547c3e5212d0f4ba3a3

That's part of it, but we also want to assert the a11y description in the error and loading cases, and we want to assert that if `NxCombobox` receives an _external_ `aria-describedby`, that it uses the id provided there in addition to the internal description elements (e.g. test the logic around the `inputDescribedBy` variable in the `NxCombobox` implementation)

Reply:
> That's part of it, but we also want to assert the a11y description in the error and loading cases, and we want to assert that if NxCombobox receives an external aria-describedby, that it uses the id provided there in addition to the internal description elements (e.g. test the logic around the inputDescribedBy variable in the NxCombobox implementation)

Updated it for different alert messages(No results found/ loading/ errors/ external message referenced with id/aria-describedby). 4d494e7350646aa88bba3067bd42bdd0b1062663. Thanks.